Packages/ClientRuntime/Sources/Waiters/Waiter.swift

```diff
@@ -0,0 +1,33 @@
+//
+// Copyright Amazon.com Inc. or its affiliates.
+// All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+/// Makes a waiter operation, as configured by the caller, into an async sequence of waiter status events.
+/// Intended to be a generic type for use when waiting on any Smithy `operation`.
+public class Waiter<Input, Output> {
+    public let input: Input
+    public let acceptors: [Acceptor<Input, Output>]
+    let minDelay: TimeInterval
+    let maxDelay: TimeInterval
+    let maximumWaitTime: TimeInterval
+    public let operation: (Input) async throws -> Output
+
+    public init(input: Input, acceptors: [Acceptor<Input, Output>], minDelay: TimeInterval, maxDelay: TimeInterval,
+                maximumWaitTime: TimeInterval, operation: @escaping (Input) async throws -> Output) {
```

Comment:
Nit: To provide a consistent format/style when using line breaks, can we put each argument on its own line. This follows the style that is used within the Swift standard library. (We should have our linter catch this as well)

```suggestion
    public init(
        input: Input, 
        acceptors: [Acceptor<Input, Output>],
        minDelay: TimeInterval, 
        maxDelay: TimeInterval,
        maximumWaitTime: TimeInterval, 
        operation: @escaping (Input) async throws -> Output
    ) {
```

Reply:
I've reformatted my method signatures this way, at least the ones with more than one param.